src/react.ts

```diff
@@ -107,8 +107,13 @@ const createImpl = <T>(createState: StateCreator<T, [], []>) => {
       "[DEPRECATED] Passing a vanilla store will be unsupported in a future version. Instead use `import { useStore } from 'zustand'`.",
     )
   }
-  const api =
+  const api = (
     typeof createState === 'function' ? createStore(createState) : createState
+  ) as WithReact<StoreApi<T>>
```

Comment:
@dai-shi I think `getServerState` should be required in `WithReact`:

https://github.com/pmndrs/zustand/blob/517524d2b05843c6a26d2713af08cc0ca1f62f3c/src/react.ts#L25-L27

but I'm not sure if that would be a breaking change ?

Reply:
Well, I think `WithReact` is for a different purpose. Anyway, the thing is that we don't want to attach `getServerState` to vanilla `api`, so that's why it doesn't require here.